pagerdutyplugin/resource_pagerduty_business_service.go

```diff
@@ -0,0 +1,230 @@
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	helperResource "github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+type resourceBusinessService struct {
+	client *pagerduty.Client
+}
+
+var (
+	_ resource.ResourceWithConfigure   = (*resourceBusinessService)(nil)
+	_ resource.ResourceWithImportState = (*resourceBusinessService)(nil)
+)
+
+func (r *resourceBusinessService) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "pagerduty_business_service"
+}
+
+func (r *resourceBusinessService) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name":     schema.StringAttribute{Required: true},
+			"id":       schema.StringAttribute{Computed: true},
+			"html_url": schema.StringAttribute{Computed: true},
+			"self":     schema.StringAttribute{Computed: true},
+			"summary":  schema.StringAttribute{Computed: true},
+			"description": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString("Managed by Terraform"),
+			},
+			"type": schema.StringAttribute{
+				Optional:           true,
+				Computed:           true,
+				Default:            stringdefault.StaticString("business_service"),
+				DeprecationMessage: "This will change to a computed resource in the next major release.",
+				Validators:         []validator.String{stringvalidator.OneOf("business_service")},
+			},
+			"point_of_contact": schema.StringAttribute{Optional: true},
+			"team":             schema.StringAttribute{Optional: true},
+		},
+	}
+}
+
+func (r *resourceBusinessService) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan resourceBusinessServiceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	businessServicePlan := buildPagerdutyBusinessService(ctx, &plan)
+
+	err := helperResource.RetryContext(ctx, 5*time.Minute, func() *helperResource.RetryError {
+		log.Printf("[INFO] Creating PagerDuty business service %s", plan.Name)
+		bs, err := r.client.CreateBusinessServiceWithContext(ctx, businessServicePlan)
+		if err != nil {
+			return helperResource.NonRetryableError(err)
+		} else if bs != nil {
+			businessServicePlan.ID = bs.ID
+		}
+		return nil
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error calling CreateBusinessServiceWithContext",
+			err.Error(),
+		)
+		return
+	}
+
+	err = helperResource.RetryContext(ctx, 5*time.Minute, func() *helperResource.RetryError {
```

Comment:
Please use refactored re-usable version of this. Please check next comment in `Read` for better understanding.